replication/logic/replication_logic.go

```diff
@@ -469,29 +483,23 @@ func (fs *Filesystem) doPlanning(ctx context.Context) ([]*Step, error) {
 			return nil, conflict
 		}
 
+		if initialSend {
+			// Add a nil path to the beginning so our loop over each step starts with a nil...FIRSTSNAPSHOT step.
+			// The nil `from` is what indicates the send is not incremental.
+			path = append([]*pdu.FilesystemVersion{nil}, path...)
+		}
+
 		steps = make([]*Step, 0, len(path)) // shadow
-		if len(path) == 1 {
+		for i := 0; i < len(path)-1; i++ {
```

Comment:
This could also be a range. Simpler and optimized for slices.

Reply:
I think I need the range in this case to get i, since I'm essentially windowing across the slice.

Reply:
Oh right, sorry I missed `path[i+1]`.